site/zenodo_rdm/openaire/utils.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""OpenAire related helpers."""
+
+import hashlib
+import urllib
+
+from flask import current_app
+from invenio_access.permissions import system_identity
+from invenio_communities.proxies import current_communities
+from invenio_vocabularies.proxies import current_service as vocab_service
+from requests import Session
+
+OPENAIRE_NAMESPACE_PREFIXES = {
+    "publication": "od______2659",
+    "dataset": "od______2659",
+    "software": "od______2659",
+    "other": "od______2659",
+}
+
+OPENAIRE_ZENODO_IDS = {
+    "publication": "opendoar____::2659",
+    "dataset": "opendoar____::2659",
+    "software": "opendoar____::2659",
+    "other": "opendoar____::2659",
+}
+
+OA_PUBLICATION = "publication"
+OA_DATASET = "dataset"
+OA_SOFTWARE = "software"
+OA_OTHER = "other"
+
+
+def get_resource_type_vocabulary(resource_type):
```

Comment:
Ideally, this should not be called many times (like it happens now), however, it happens. If this is not cached on the vocabularies service, I can come back to this afterwards.